### PR TITLE
[dv] Update VCS cov merge opts

### PR DESCRIPTION
**Pull request by sriyerg**
- Enable tests to be shown in coverage reports, with grading data
showing the % improvement contributed by each test - useful to look at
and prune the regression footprint.
- Enable merging of assertion coverage across VDBs
- Enable merging of covergroups across VDBs in union mode.
- Enable LCA features (does not require extra license)

Signed-off-by: Srikrishna Iyer <sriyer@google.com>

### CODE REVIEW

**Review by rswarbrick — APPROVED** [review]
Looks good to me, and the functional coverage counting changes make much more sense, thanks.

Does this have an impact on blocks that use Xcelium for signoff? Or is that already doing the more sensible calculation?

**Review by rasmus-madsen — APPROVED** [review]
LGTM
Will you update for xcelium also  or do you want me to do that?

**Review by udinator — APPROVED** [review]

File: hw/dv/tools/dvsim/vcs.hjson [hunk 1]
```diff
@@ -125,31 +125,45 @@
...
-                        "-group instcov_for_score",
-                        "-line nocasedef",
-                        "-format both",
-                        "-elfile {vcs_cov_excl_files}",
-                        "-report {cov_report_dir}"]
-  cov_report_txt:       "{cov_report_dir}/dashboard.txt"
-  cov_report_page:      "dashboard.html"
+  cov_report_dir:   "{scratch_path}/cov_report"
+  cov_report_cmd:   "{job_prefix} urg"
+  cov_report_opts:  ["-lca",
```
Comment (matutem) [inline hw/dv/tools/dvsim/vcs.hjson:148]: There is also cost-based grading, which can be handy. For the grader I wrote a few years ago I used "-grade cost clocktime", so tests that achieve the highest coverage in the least wall-clock time are preferred. I also added "testfile" after "clocktime", per the urg manual:
"To generate the gradedtests.txt test file, specify the testfile argument with the –grade option..."

I also recall grading could take lots of time, so we had to figure out which set of options was preferable. I can do an archaeological dive if this becomes an issue.

**Review by matutem — APPROVED** [review]
Please note the comment about cost-based grading. Perhaps can be used in some subsequent PRs, once we see how this is working.

**Review by cindychip — APPROVED** [review]

**Review by weicaiyang — APPROVED** [review]
LGTM

**Comment by sriyerg** [comment]
> Looks good to me, and the functional coverage counting changes make much more sense, thanks.
> 
> Does this have an impact on blocks that use Xcelium for signoff? Or is that already doing the more sensible calculation?

No impact on Xcelium. I an not an expert on it so I will let @rasmus-madsen figure out if the options need to change.

**Comment by sriyerg** [comment]
> LGTM
> Will you update for xcelium also or do you want me to do that?

I will let you take care of it , thanks!

File: hw/dv/tools/dvsim/vcs.hjson (hunk 1 above)
Comment (sriyerg) [reply hw/dv/tools/dvsim/vcs.hjson:148]: Agree - lets fine tune in multiple passes.

**Pull request merged**